src/emulator/gui/src/settings_dialog.cpp

```diff
@@ -0,0 +1,105 @@
+// Vita3K emulator project
+// Copyright (C) 2019 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include "private.h"
+#include <../../config.h>
```

Comment:
This is hacky, we should have config stuff in a separate module instead of in `emulator`.
The other alternative would be to have `gui` access `emulator` headers if that's possible, but it's probably a worse idea.
@petmac Should we accept this line?

Reply:
He includes host/state.h, shouldn't Config be included anyway?

Reply:
Citing https://github.com/Vita3K/Vita3K/blob/969559a9fc60a698112e444493942f12d65e211b/src/emulator/host/include/host/state.h#L20

Reply:
Config stuff is not in `host`, it's in `emulator` atm (the root dir).

Reply:
Oh yeah you can include emulator stuff everywhere apparently.
host just does `#include "config.h"` so @Zangetsu38 do that here too.

Reply:
I don't see the need for this line if it's already being included through host/state.h though (he includes it just a few lines later). Can we cut it?

Reply:
No, if `state.h` stops including it for some reason that code will break. Each file should include whatever is uses, we shouldn't rely on hidden inclusions like that.

Reply:
Config.h is needed for config::host.... look end file and is Visual studio have set it like that i have not succès set normally ^^

Reply:
...did you try `#include "config.h"`?

Reply:
Hmm not remember but tester <with > no have works

Reply:
I test after

Reply:
Got it. Makes sense.